packages/discord.js/src/managers/GuildChannelManager.js

```diff
@@ -99,7 +99,7 @@ class GuildChannelManager extends CachedManager {
   /**
    * Options used to create a new channel in a guild.
    * @typedef {CategoryCreateChannelOptions} GuildChannelCreateOptions
-   * @property {CategoryChannelResolvable} [parent] Parent of the new channel
+   * @property {CategoryChannelResolvable|null} [parent] Parent of the new channel
```

Comment:
```suggestion
   * @property {?CategoryChannelResolvable} [parent] Parent of the new channel
```

Reply:
is this correct though? The property is already optional, why would you pass it as null?

Reply:
To unset the parent

Reply:
Setting it to null clears it while omitting it or setting it to undefined just ignores it. For example, if you wanted to delete the current topic by passing in null, you can't do it, at the very least, not without typescripts errors, even though thats how it's supposed to be done.

Reply:
clears what? This is the typedef for creating a channel. You can't clear the parent of a channel you just created, that doesn't make sense

Reply:
I believe the same point may be made to the above review about a channel's topic during guild creation additionally?

Reply:
yes, I only commented on one but it applies to the entire PR tbh

Reply:
Yes, but for a different reason. That one is `PartialChannelData` not `CategoryCreateChannelOptions`, but we don't explicitly set properties to null in a Partial. It's just there or it isn't.

Reply:
There's been a misunderstanding on my side. Here's a better example, a non-partial non-category channel's parent value is a CategoryChannel or it's null. If we wanted to save a channel as JSON and use it to create new channels we face the issue that parent accepts CategoryChannelResolvable | undefined. Of which our channel parent value might be neither because it could be null. Sure you could just transform it to undefined but to me it doesn't really make sense to not allow null (because it works and it's accepted by Discord API). It just seems inconsistent. As for the channel topic, as of now, there isn't a way to clear a topic programmatically according to discord.js typings because it doesn't accept null. (setting it to undefined doesn't remove it).

Reply:
`BaseGuildTextChannel#setTopic` does accept null: <https://github.com/discordjs/discord.js/blob/fdeac9d9fba06c532eca296ddd8479047bc732bf/packages/discord.js/typings/index.d.ts#L507>

Arguably, you could also change `ChannelData` to accept `string | null`, as this is the correct typedef for the `GuildChannel#edit` method. `PartialChannelData` is only used when creating channels in the `GuildCreateOptions` payload, so `null` does not belong here.

> Sure you could just transform it to undefined but to me it doesn't really make sense to not allow null 

`null` and `undefined` are not the same thing. Trying to treat them as the same thing would be inconsistent, not the other way around. Transform it.